x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/SslConfig.java

```diff
@@ -62,8 +63,19 @@
 
     private final SSLContext sslContext;
 
-    SslConfig(Properties settings) {
-        enabled = StringUtils.parseBoolean(settings.getProperty(SSL, SSL_DEFAULT));
+    SslConfig(Properties settings, URI baseURI) {
+        boolean isSchemaPresent = baseURI.getScheme() != null;
+        boolean isSSLPropertyPresent = settings.getProperty(SSL) != null;
+        boolean isHttpsScheme = "https".equals(baseURI.getScheme());
+        
+        if (!isSSLPropertyPresent && !isSchemaPresent) {
+            enabled = StringUtils.parseBoolean(SSL_DEFAULT);
+        } else {
+            if (isSSLPropertyPresent && isHttpsScheme && !StringUtils.parseBoolean(settings.getProperty(SSL))) {
```

Comment:
I think you can remove isSSLPropertyPresent here, it's always true as it's checked above.

Reply:
It's not necessarily always true @matriv. The first condition could be `false` if `isSSLPropertyPresent` is also `false` (but `isSchemaPresent` true). This check is on purpose set like this because I want the exception to be thrown only in that specific case.